src/librawspeed/decoders/NakedDecoder.cpp

```diff
@@ -44,50 +44,55 @@ NakedDecoder::NakedDecoder(FileMap* file, Camera* c) :
 
 NakedDecoder::~NakedDecoder() = default;
 
-RawImage NakedDecoder::decodeRawInternal() {
-  uint32 width=0, height=0, filesize=0, bits=0, offset=0;
-  if(cam->hints.find("full_width") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_width"))->second;
-    width = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find width");
-
-  if(cam->hints.find("full_height") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("full_height"))->second;
-    height = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find height");
-
-  if(cam->hints.find("filesize") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("filesize"))->second;
-    filesize = (uint32) atoi(tmp.c_str());
-  } else
-    ThrowRDE("Naked: couldn't find filesize");
-
-  if(cam->hints.find("offset") != cam->hints.end()) {
-    string tmp = cam->hints.find(string("offset"))->second;
-    offset = (uint32) atoi(tmp.c_str());
-  }
+static const map<string, BitOrder> order2enum = {
+    {"plain", BitOrder_Plain},
+    {"jpeg", BitOrder_Jpeg},
+    {"jpeg16", BitOrder_Jpeg16},
+    {"jpeg32", BitOrder_Jpeg32},
+};
+
+void NakedDecoder::parseHints() {
```

Comment:
I also think your new code is better than the old! I just think this one aspect of increasing the scope of the code and the involved variables is not the best idea. One alternative to show the code reader/developer that this first part is basically "parseHint" could be to simply document it that way? Or maybe move it in a separate function but pass it the parameters it needs instead of putting them in the class scope. Then you have a separate entity without the scoping issue.

I _really_ had to wade through a lot of this type of code in the Tiff related classes where it was all about first getting an idea of which variables/functions are actually used/modified where and affect each other in what way. I'm saying this is a real issue and it makes the code as a whole _less_ easy to understand.

Of course, big spagetti-code functions are not what I'm after ;).